src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelist.java

```diff
@@ -113,6 +117,10 @@ public final void reset(Collection<? extends Whitelist> delegates) {
             for (ProxyWhitelist pw : wrappers.keySet()) {
                 pw.reset();
             }
+            if (this.wrappers.isEmpty()) {  // Top-level ProxyWhitelist should be the only cache
+                // Top-level ProxyWhitelist should precache the statically permitted signatures
+                ((EnumeratingWhitelist)(this.delegates.get(0))).precache();
+            }
```

Comment:
**TODO** we need a trivial test that the ProxyWhitelist caching actually correctly caches where it should.